atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -591,6 +592,15 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
                         runtime.map(AtlasDbRuntimeConfig::compact)),
                 closeables);
 
+        Set<LockWatchReference> lockWatchesToRegister = schemas().stream()
+                .map(Schema::getLockWatches)
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
+
+        if (!lockWatchesToRegister.isEmpty()) {
+            lockAndTimestampServices.lockWatcher().registerPreciselyWatches(lockWatchesToRegister);
+        }
```

Comment:
Please add javadoc either here, or preferably on the method in LWM, that this will be overwritten if anyone else calls the `registerPreciselyWatches` method. It might be obvious, but I'd prefer that we document it so we don't shoot ourselves in the foot.

Reply:
👍